File: Templates/Java/integrationtesting/src/main/java/com/example/integrationtesting/JsonPlaceholderClientImpl.java
package com.example.integrationtesting;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestClient;

@Component
public class JsonPlaceholderClientImpl implements JsonPlaceholderClient {

    private final RestClient client;
    public JsonPlaceholderClientImpl(@Value("${httpClientSettings.baseUrl}") String baseUrl) {
        client = RestClient.builder().baseUrl(baseUrl).build();
    }

    public String get(String path, Optional<String> queryString) {
        var url = path + queryString.orElse("");
        var response = client.get().uri(url).retrieve().body(String.class);
        return response;
    }
}

File: Templates/Kotlin/integrationtesting/src/main/java/com/example/integrationtesting/JsonPlaceholderClientImpl.kt
package com.example.integrationtesting

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.springframework.web.client.RestClient

@Component
class JsonPlaceholderClientImpl(@Value($$"${httpClientSettings.baseUrl}") baseUrl: String) : JsonPlaceholderClient {
    private val client: RestClient = RestClient.builder().baseUrl(baseUrl).build()

    override fun get(path: String?, queryString: String?): String? {
        val url = path + (queryString ?: "")
        val response = client.get().uri(url).retrieve().body(String::class.java)
        return response
    }
}
